services/server/server_linux.go

```diff
@@ -37,6 +38,20 @@ func apply(ctx context.Context, config *srvconfig.Config) error {
 		}
 	}
 	if config.Cgroup.Path != "" {
+		if cgroups.Mode() == cgroups.Unified {
+			cg, err := cgroupsv2.LoadManager("/sys/fs/cgroup", config.Cgroup.Path)
+			if err != nil {
+				if err != cgroupsv2.ErrCgroupDeleted {
+					return err
+				}
+				if cg, err = cgroupsv2.NewManager("/sys/fs/cgroup", config.Cgroup.Path, nil); err != nil {
+					return err
+				}
+			}
+			if err := cg.AddProc(uint64(os.Getpid())); err != nil {
+				return err
+			}
+		}
```

Comment:
it should be `if else`?

Reply:
Note just to eliminate these potentially error prone `if...else` clauses, maybe something like a cgroup loader interface is needed.

Reply:
thanks, added `else`.

Adding a loader interface isn't straightforward due to the design differences across v1 and v2.